accepted/ddlm_dictionary_style_guide.md

```diff
@@ -250,23 +250,70 @@ compound objects.
 
 ### 2.5 Enumeration ranges
 
-When giving enumeration ranges for `Integer` data items, no decimal
-point should be used. When giving enumeration ranges for `Real` data
-items, a decimal point should be used with at least one digit after
-the point. Any following digits, if present, must be non-zero.
+Values of the `_enumeration.range` attribute should be expressed in a format that best reflects the content type of the defining item.
+That is, numeric range limits of data items with the `Integer` content type should be formatted as integers while data items with the `Real` content type should be formatted as floating-point real numbers.
+Additional formatting rules for enumeration ranges are provided in Section 2.5.1 and Section 2.5.2.
 
-#### Examples
+#### 2.5.1 Integer ranges
+
+Numeric range limits of data items with the `Integer` content type should be expressed as integers that:
+1. Do not include non-significant leading zeros, e.g. '7' instead of '007'.
+2. Do not include a fractional part, e.g. '1' instead of '1.0'.
+3. Do not include a trailing decimal separator, e.g. '2' instead of '2.'.
+4. Do not include the '+' symbol, e.g. '42' instead '+42'.
+5. Do not include a signed zero, e.g. '0' instead of '+0' or '-0'.
 
+The following regular expression may be used to check if a number adheres to the integer range limit formatting rules:
+```
+^
+(
+  0|( [-]?[1-9][0-9]* )
+)
+$
+```
+The regular expression above is formatted for readability using the additional syntax rules enabled by the `/x` Perl regular expression modifier (e.g. any unescaped whitespace symbols must be ignored).
+
+##### Examples of properly formatted integer number ranges
 ```
-# Integer ranges
 1:230
 0:
+:27
 -8:8
+```
+
+### 2.5.2 Real number ranges
 
-# Real number ranges
+Numeric range limits of data items with the `Real` content type should be expressed using floating-point real numbers that:
+1. Include at least one digit before the decimal separator, e.g. '0.5' instead of '.5'.
+2. Include at least one digit after the decimal separator, e.g. '7.0' instead of '7.' or '7'.
+3. Include the smallest number of non-significant leading zeros that still satisfies other formatting rules, e.g. '0.25' instead of '000.25'.
+4. Include the smallest number of non-significant trailing zeros that still satisfies other formatting rules, e.g. '13.0' instead of '13.000'.
+5. Do not include the '+' symbol, e.g. '42.0' instead '+42.0'.
+6. Do not include a signed zero, e.g. '0.0' instead of '+0.0' or '-0.0'.
+
+The following regular expression may be used to check if a number adheres to the real number range limit formatting rules:
+
+```
+^
+(
+  # Real number '0.0'.
+  ( 0[.]0 ) |
+  # All integer-like numbers, e.g. '-5.0'.
+  ( [-]?([1-9][0-9]*)[.]0 ) |
+  # All remaining floating-point numbers.
+  ( [-]?(0|([1-9][0-9]*))[.]([0-9]*[1-9]) )
```

Comment:
Currently, here and in the validator I follow the CIF_1.1 interpretation of an integer which does not allow exponents (see the <Integer> construct in [1]) since similar information is not available for CIF_2.0. Under the classification of CIF_1.1 1 value '1E+2' is a float and not an integer. The review copy of one upcoming chapter mentions section 1.2.4.2.3 that further clarifies the interpretation of float and integer content types in the context of CIF_2.0, but I do not have access to it (maybe @jamesrhester can shed some light on this)?

However, I do agree that fully typing out '1E+100' does sound unreasonable therefore we may allow a subset of numbers with exponents that are numerically equivalent to integers. I do imagine that it might sometimes be more convenient to specify a range as `1E+0:1E+5` than `1:100000`.

That being said, there currently aren't any range limits of that magnitude in the dictionaries  so we might want to delay the introduction of exponents until there is a specific use case. Furthermore, I am not sure if *simple* regular expressions are capable of distinguishing between exponents that are equivalent to integers and the one that are not unless we also introduce some additional restriction on the exponents, e.g. do not permit the fractional numbers. That is, we would require '3.14E+5' to be written as '314E+3' which might look a bit strange. If this is not desired, I am also fine with doing away with the regex altogether.

@jamesrhester , @rowlesmr what is your preference? Should we wait until there is an actual use case for exponents or should we cover our bases in this PR? 

[1] https://www.iucr.org/resources/cif/spec/version1.1/cifsyntax

Reply:
I would leave out exponents until there is a need for them.

Reply:
Leave them out. You would probably need a proper parser to distinguish exponentiated integers from floats.

Reply:
Ok.